sdk/appconfiguration/app-configuration/types/app-configuration.d.ts

```diff
@@ -0,0 +1,387 @@
+     * labels and accept datetime.
+     *
+     * Example code:
+     * ```ts
+     * const revisionsIterator = await client.listRevisions({ keys: ["MyKey"] });
+     * ```
+     * @param options Optional parameters for the request.
+     */
+    listRevisions(options?: ListRevisionsOptions): PagedAsyncIterableIterator<ConfigurationSetting, ListRevisionsPage>;
+    private getListRevisionsIterator;
+    private listRevisionsByPage;
+    /**
+     * Sets the value of a key in the Azure App Configuration service, allowing for an optional etag.
+     * @param key The name of the key.
+     * @param configurationSetting A configuration value.
+     * @param options Optional parameters for the request.
+     *
+     * Example code:
+     * ```ts
+     * await client.setConfigurationSetting({ key: "MyKey", value: "MyValue" });
+     * ```
+     */
+    setConfigurationSetting(configurationSetting: SetConfigurationSettingParam, options?: SetConfigurationSettingOptions): Promise<SetConfigurationSettingResponse>;
+    /**
+     * Sets a key's value to read only
+     * @param id The id of the configuration setting to set to read-only.
+     */
+    setReadOnly(id: ConfigurationSettingId, options?: SetReadOnlyOptions): Promise<SetReadOnlyResponse>;
+    /**
+     * Makes the key's value writable again
+     * @param id The id of the configuration setting to make writable.
+     */
+    clearReadOnly(id: ConfigurationSettingId, options?: ClearReadOnlyOptions): Promise<ClearReadOnlyResponse>;
+}
+
+/**
+ * Options for clearReadOnly
+ */
+export declare interface ClearReadOnlyOptions extends HttpOnlyIfUnchangedField, RequestOptionsBase {
+}
+
+/**
+ * Response when clearing the read-only status from a value
+ */
+export declare interface ClearReadOnlyResponse extends ConfigurationSetting, SyncTokenHeaderField, HttpResponseField<SyncTokenHeaderField> {
+}
+
+/**
+ * Configuration setting with extra metadata from the server, indicating
+ * its etag, whether it is currently readOnly and when it was last modified.
+ */
+export declare interface ConfigurationSetting extends ConfigurationSettingParam {
+    /**
+     * Whether or not the setting is read-only
+     */
+    readOnly: boolean;
+    /**
+     * The date when this setting was last modified
+     */
+    lastModified?: Date;
+}
+
+/**
+ * Fields that uniquely identify a configuration setting
+ */
+export declare interface ConfigurationSettingId {
+    /**
+     * The key for this setting
+     */
+    key: string;
+    /**
+     * The label for this setting. Leaving this undefined means this
+     * setting does not have a label.
+     */
+    label?: string;
+    /**
+     * The etag for this setting
+     */
+    etag?: string;
+}
+
+/**
+ * Necessary fields for updating or creating a new configuration setting
+ */
+export declare interface ConfigurationSettingParam extends ConfigurationSettingId {
+    /**
+     * The content type of the setting's value
+     */
+    contentType?: string;
```

Comment:
is this a MIME type? If so maybe `contentMimeType` ?

Reply:
It's actually a totally free-form content type (no restrictions that I see).

They use it in a few ways themselves to denote content formats that are "internal" to the API, like feature flags and (as @willmtemple showed me the other day) linking to a KeyVault setting.

Reply:
it also carries information about the encoding. it's supposed to be an HTTP Content-Type formatted string but as far as I can tell you can put anything into the box in the portal